lib/MusicBrainz/Server/Edit/Release/ReorderMediums.pm

```diff
@@ -103,10 +104,28 @@ sub build_display_data {
 
 sub accept {
     my ($self) = @_;
-    $self->c->model('Medium')->reorder(
-        map {
-            $_->{medium_id} => $_->{new}
-        } @{ $self->data->{medium_positions} });
+
+    my %medium_positions = map { $_->{medium_id} => $_->{new} }
+        @{ $self->data->{medium_positions} };
+    my $new_positions = [values %medium_positions];
+
+    my $possible_conflicts =
+        $self->c->sql->select_list_of_hashes(<<'EOSQL', $self->release_id, $new_positions);
+            SELECT id, position
+              FROM medium
+             WHERE release = ?
+               AND position = any(?)
+EOSQL
+
+    for my $row (@$possible_conflicts) {
+        unless (exists $medium_positions{$row->{id}}) {
+            MusicBrainz::Server::Edit::Exceptions::FailedDependency->throw(
+                'Can’t move a medium into position ' . $row->{position} . ', where one already exists.'
+            );
```

Comment:
Right, ModBot will catch the exception and do what's necessary. The basic flow of operations is this:

1. admin/CheckVotes.pl ("ModBot") is run hourly on astro, via admin/cron/hourly.sh
2. `MusicBrainz::Server::EditQueue::process_edits` is called from CheckVotes.pl
3. `process_edits` will loop over open/expired edits, and try to accept relevant ones [here](https://github.com/mwiencek/musicbrainz-server/blob/1b01b1f/lib/MusicBrainz/Server/EditQueue.pm#L160).
4. That'll invoke the `accept` method on the edit, the one I'm changing here, [within a try/catch block](https://github.com/mwiencek/musicbrainz-server/blob/1b01b1f/lib/MusicBrainz/Server/Data/Edit.pm#L682).
5. The `FailedDependency` exception [is caught](https://github.com/mwiencek/musicbrainz-server/blob/1b01b1f/lib/MusicBrainz/Server/Data/Edit.pm#L687), and a note containing the error message is left by the ModBot editor account.